lib/filter-schema.js

```diff
@@ -109,12 +109,31 @@ async function filterSchema (graphQLSchema, policies, context) {
             operation: { kind: 'OperationDefinition', operation: 'query' },
             variableValues: {}
           }
-          // The null parameters are: https://graphql.org/learn/execution/#root-fields-resolvers
-          // - parent: it is not possible know it since the resolver is not executed yet
-          // - args: it is not expected that the introspection query will have arguments for the directives policies
-          canShowDirectiveField = await policyFunction(fieldPolicy, null, null, context, info)
-          if (canShowDirectiveField instanceof Error || !canShowDirectiveField) {
-            canShowDirectiveField = false
+
+          if (Array.isArray(fieldPolicy)) {
```

Comment:
This check should be done here too

https://github.com/mercurius-js/auth/blob/main/lib/auth.js#L113

It uses the policy structure to add the logic.

To spot this issue, you need to add a simple usage test of the repeatable directive that hit a real query a not the introspection one

Reply:
I've fixed the method and added some extra test cases. Thanks for pointing me in the right direction.